src/routes/subscriptions/subscription.controller.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { EmailControllerModule } from '@/routes/email/email.controller.module';
+import { EmailApiModule } from '@/datasources/email-api/email-api.module';
+import { TestEmailApiModule } from '@/datasources/email-api/__tests__/test.email-api.module';
+import { EmailDataSourceModule } from '@/datasources/email/email.datasource.module';
+import { TestEmailDatasourceModule } from '@/datasources/email/__tests__/test.email.datasource.module';
+import { CacheModule } from '@/datasources/cache/cache.module';
+import { TestCacheModule } from '@/datasources/cache/__tests__/test.cache.module';
+import { RequestScopedLoggingModule } from '@/logging/logging.module';
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { NetworkModule } from '@/datasources/network/network.module';
+import { TestNetworkModule } from '@/datasources/network/__tests__/test.network.module';
+import { IEmailDataSource } from '@/domain/interfaces/email.datasource.interface';
+import { TestAppProvider } from '@/__tests__/test-app.provider';
+import * as request from 'supertest';
+import { faker } from '@faker-js/faker';
+import { Subscription } from '@/domain/email/entities/subscription.entity';
+
+describe('Subscription Controller tests', () => {
+  let app;
+  let emailDatasource;
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    jest.useFakeTimers();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule.register(configuration), EmailControllerModule],
+    })
+      .overrideModule(EmailApiModule)
+      .useModule(TestEmailApiModule)
+      .overrideModule(EmailDataSourceModule)
+      .useModule(TestEmailDatasourceModule)
+      .overrideModule(CacheModule)
+      .useModule(TestCacheModule)
+      .overrideModule(RequestScopedLoggingModule)
+      .useModule(TestLoggingModule)
+      .overrideModule(NetworkModule)
+      .useModule(TestNetworkModule)
+      .compile();
+
+    emailDatasource = moduleFixture.get(IEmailDataSource);
+
+    app = await new TestAppProvider().provide(moduleFixture);
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('deletes category successfully', async () => {
+    const subscriptionKey = faker.word.sample();
+    const subscriptionName = faker.word.sample(2);
+    const token = faker.string.uuid();
+    const subscriptions = [
+      {
+        key: subscriptionKey,
+        name: subscriptionName,
+      },
+    ] as Subscription[];
+    emailDatasource.unsubscribe.mockResolvedValueOnce(subscriptions);
+
+    await request(app.getHttpServer())
+      .delete(`/v1/subscriptions/?category=${subscriptionKey}&token=${token}`)
+      .expect(200)
+      .expect({});
+
+    expect(emailDatasource.unsubscribe).toHaveBeenCalledWith({
+      categoryKey: subscriptionKey,
+      token: token,
+    });
+    expect(emailDatasource.unsubscribeAll).toHaveBeenCalledTimes(0);
```

Comment:
Nit: can we also check that it no longer exists in the database?

Reply:
The test setup does not require a database (it's replaced with a fake). So that assertion would not work here because there's no database.

However that integration is tested in `src/datasources/email/email.datasource.spec.ts`.

The goal of this test is to cover everything up until the database call. Database calls are then covered in the test that I mentioned.